lib/client.c

```diff
@@ -432,14 +438,45 @@ _snowflake_check_connection_parameters(SF_CONNECT *sf) {
         // construct a host parameter if not specified,
         char buf[1024];
         if (sf->region) {
-            sf_sprintf(buf, sizeof(buf), "%s.%s.snowflakecomputing.com",
-                     sf->account, sf->region);
+            if (strncasecmp(sf->region, "cn-", 3) == 0)
+            {
+                //region started with "cn-", use "cn" for top domain
+                sf_sprintf(buf, sizeof(buf), "%s.%s.snowflakecomputing.cn",
+                         sf->account, sf->region);
+            }
+            else
+            {
+                sf_sprintf(buf, sizeof(buf), "%s.%s.snowflakecomputing.com",
+                         sf->account, sf->region);
+            }
         } else {
             sf_sprintf(buf, sizeof(buf), "%s.snowflakecomputing.com",
                      sf->account);
         }
         alloc_buffer_and_copy(&sf->host, buf);
     }
+
+    char* top_domain = strrchr(sf->host, '.');
+    if (top_domain)
+    {
+        top_domain++;
+    }
+    else
+    {
+        // It's basically impossible not finding top domain in host.
+        // Log the entire host just in case.
+        top_domain = sf->host;
+    }
+
+    if (strcasecmp(top_domain, "com") == 0)
```

Comment:
please change the log to expose CHINA and GLOBAL only like in JDBC https://github.com/snowflakedb/snowflake-jdbc/blob/60e9e541794bbd80a33b00cf8039c8512d8c5361/src/main/java/net/snowflake/client/core/SFSession.java#L625-L627

Reply:
done